docs/snapcraft-syntax.md

```diff
@@ -21,6 +21,8 @@ contain.
   The type of confinement supported by the snap. Can be either "devmode" (i.e.
   this snap doesn't support running under confinement) or "strict" (i.e. full
   confinement supported via interfaces).
+* `assumes` (list of strings)
+  A feature tag list that the core needs to support for this snap to be run.
```

Comment:
From the bug, this seems to be an install-time check. Maybe "A list of features that must be supported by the core in order for this snap to install." ?